redash/models.py

```diff
@@ -715,6 +715,32 @@ def recent(cls, groups, user_id=None, limit=20):
 
         return query
 
+    @classmethod
+    def fork(cls, id, user, org):
```

Comment:
I think that fork logic (fetch query and chart data by query id, then change their name and id, then insert them) should be model's method.